paddle/framework/op_registry.h

```diff
@@ -75,6 +81,20 @@ class OpRegistry {
   static std::unique_ptr<OperatorBase> CreateOp(const OpDesc& op_desc);
 
   static std::unique_ptr<OperatorBase> CreateGradOp(const OperatorBase& op);
+
+  // compile time InferShape
+  static void InferShape(const OpDesc& op_desc,
```

Comment:
I see. We might need to change this PR according to the recently merged design of [`ProgramDesc`](https://github.com/PaddlePaddle/Paddle/pull/4241).